### ISSUE TITLE
NameError: name 'sqlcur' is not defined during "gyb estimate"

### ISSUE THREAD

**Issue by bfloydd** [jay0lee/got-your-back#59]
gyb --email xxx@xxx.com --action estimate

Got 83668 Message IDs
GYB needs to examine 83668 messages
Traceback (most recent call last):
  File "<string>", line 1491, in <module>
  File "<string>", line 1454, in main
NameError: name 'sqlcur' is not defined

Tried on different computers in cmd, gitbash, no luck

Python version 3.4.3

**Comment by mciantyre** [comment]
Same issue here, same version.

**Comment by MaffooBristol** [comment]
Same

**Comment by turbo2ltr** [comment]
me too.

```.......
Got 136200 Message IDs
Got 136300 Message IDs
Got 136400 Message IDs
Got 136500 Message IDs
Got 136506 Message IDs

GYB needs to examine 136506 messages
Traceback (most recent call last):
  File "<string>", line 1491, in <module>
  File "<string>", line 1454, in main
NameError: name 'sqlcur' is not defined
```


```
gyb --version
Got Your Back 0.41
http://git.io/gyb
Jay Lee - jay0lee@gmail.com
Python 3.4.3 64-bit final
Windows-7-6.1.7601-SP1 AMD64
```

**Comment by jonegerton** [comment]
Same for me:

Authentication successful.
Got 3 Message IDs
GYB needs to examine 3 messages
Traceback (most recent call last):
  File "<string>", line 1491, in <module>
  File "<string>", line 1454, in main
NameError: name 'sqlcur' is not defined

C:\gyb>gyb --version
Got Your Back 0.41
http://git.io/gyb
Jay Lee - jay0lee@gmail.com
Python 3.4.3 64-bit final
Windows-10.0.10240 AMD64

**Comment by jonegerton** [comment]
Looking at the source, the statements that declare `sqlcur` and `sqlcon` at lines 910 and 911 are inside the if statement `if options.action not in ['estimate'] or os.path.isfile(sqldbfile):` at line 908.

So when doing an estimate these variables are never declared.

**Comment by oakleyding** [comment]
I got these two issues too. And after patched as @jonegerton said, still get error at line 300.
I think it because the lastcheck.txt is empty. It can be fixed after `date +%s > lastcheck.txt`.

**Comment by haemi** [comment]
facing the same issue... are there any updates on this?

**Comment by dougjuh** [comment]
I made a few simple changes to my copy of gyb.py which solved the 'sqlcur' errors for me.

Change line 892 from:
    if options.action in ['backup',]:
to:
    if options.action in ['backup', 'estimate']:

Change line 902 from:
    (options.action in ['backup', 'restore-mbox'])
to:
    (options.action in ['backup', 'restore-mbox', 'estimate'])

Change line 908 from:
    if options.action not in ['estimate'] or os.path.isfile(sqldbfile):
to:
    if newDB or os.path.isfile(sqldbfile):

See also my bug fixes to the 'str' errors in issue #57.

Jay Lee said version 0.41 is still "beta" after his big overhaul, so a few bugs like this are expected, especially since they only seem to affect those who are just starting with a fresh copy of GYB.

**Comment by samsar777** [comment]
How to get this patch for windows? I use GYB 0.41 prepared, not source code.
Sorry, but have no clue how to run source code in win... and have only a week to secure my emails as my acount will be deleted (it is Gmail for ISP account).

Please help.

**Comment by samsar777** [comment]
I have found that removing white spaces from labels doesn't throw this exception.

**Comment by stevemuench** [comment]
Thanks, this fixed the issues I was having, too.

**Comment by maltokyo** [comment]
Fixed for me too. Thank you!

**Comment by jay0lee** [comment]
Fixed in master (soon to be released as 0.44)
